### ISSUE TITLE
[Feature requests] support Windows Junction

### ISSUE THREAD

**Issue by chuwza** [mikiher/audiobookshelf-windows#11]
Hi, my NAS system is Windows and I'm not using Raid, I'm using Junction to aggregate multiple hard drive directories into one directory. (C:/mnt)

Now I installed audiobookshelf-windows and I found that
audiobookshelf-windows doesn't recognize Junction, I hope it will support it, I think it's a very basic feature, Plex, Emby, Jellyfin support it.
![屏幕截图 2024-03-03 075049](https://github.com/mikiher/audiobookshelf-windows/assets/30049891/991ef28e-253e-40b0-85e5-1e67832b57c7)

(Such as picture, I'm talking about Hard Link, Symbolic Link, Junciton)
![屏幕截图(407)z](https://github.com/mikiher/audiobookshelf-windows/assets/30049891/27141c94-700c-44f1-bd00-93c613504e9d)

**Comment by mikiher** [comment]
Sorry, can you describe exactly what you were trying to do, and in what way it failed?

**Comment by chuwza** [comment]
> Sorry, can you describe exactly what you were trying to do, and in what way it failed?

Hi, do you know about symbolic links? I'm using Junction, which is similar to symbolic links, to link folders from D, E, F, and G disks under the same folder (C:/mnt) , as a way to access files from all 4 of my disks through a single shared folder (C:/mnt) . 
![屏幕截图 2024-03-04 084410](https://github.com/mikiher/audiobookshelf-windows/assets/30049891/956c8b2a-845e-4eb9-bf0a-ebc79cd71ca4)

But audioshelfbook doesn't recognize the folders linked via Junction, as shown in the picture, except for "App", which is a regular folder that I created directly in this folder, all other folders linked by Junction are not recognized.
![屏幕截图 2024-03-03 075049](https://github.com/mikiher/audiobookshelf-windows/assets/30049891/1d379cf0-84da-4a59-8b6c-31c41f37541b)

**Comment by mikiher** [comment]
Yes, I know what symlinks are. I was just trying to understand where exactly you saw the failure. 

From the image you sent, I now understand that you don't see junction links in the folder selection dialog when you create/edit a library.

Symbolic link support seems to be an issue on the server on a deeper level, and there's already an [open bug](https://github.com/advplyr/audiobookshelf/issues/2163) about it. I'm not sure how hard it is to solve - I'll try to find out. When/if that bug is resolved, I'll also try to fix the folder selection issue you reported in this bug.

As a workaround, you can point the library at the real folder location - I know this is not what you want, but this will at least work.

**Comment by chuwza** [comment]
> Yes, I know what symlinks are. I was just trying to understand where exactly you saw the failure.
> 
> From the image you sent, I now understand that you don't see junction links in the folder selection dialog when you create/edit a library.
> 
> Symbolic link support seems to be an issue on the server on a deeper level, and there's already an [open bug](https://github.com/advplyr/audiobookshelf/issues/2163) about it. I'm not sure how hard it is to solve - I'll try to find out. When/if that bug is resolved, I'll also try to fix the folder selection issue you reported in this bug.
> 
> As a workaround, you can point the library at the real folder location - I know this is not what you want, but this will at least work.

OK, Thank you very much!
